ddev/src/ddev/cli/env/test.py

```diff
@@ -105,17 +105,20 @@ def test_command(
             except json.JSONDecodeError:
                 app.abort(f'Failed to parse environments for `{integration.name}`:\n{repr(env_data_output)}')
 
+        no_python_filter = python_filter is None
+        all_environments = environment == 'all'
+
         env_names = [
             name
             for name, data in environments.items()
             if data.get('e2e-env', False)
             and (not data.get('platforms') or app.platform.name in data['platforms'])
-            and (python_filter is None or data.get('python') == python_filter)
-            and (name == environment or environment == 'all')
+            and (no_python_filter or data.get('python') == python_filter)
```

Comment:
```suggestion
            and (data.get('python', None) == python_filter)
```
Can we simplify it like this?

Reply:
Reverted the changes to simplify this. Will create a separate PR that includes the improvement on the hatch command.